src/hotspot/cpu/ppc/vm_version_ppc.cpp

```diff
@@ -305,10 +305,13 @@ void VM_Version::initialize() {
     FLAG_SET_DEFAULT(UseAESCTRIntrinsics, false);
   }
 
-  if (UseGHASHIntrinsics) {
+  if (!VM_Version::has_vsx()) {
     warning("GHASH intrinsics are not available on this CPU");
     FLAG_SET_DEFAULT(UseGHASHIntrinsics, false);
   }
+  else if (UseGHASHIntrinsics) {
+    FLAG_SET_DEFAULT(UseGHASHIntrinsics, true);
+  }
```

Comment:
This looks broken. Please take a look how it is implemented for `UseAES`.

Reply:
hotspot uses 2 spaces indentation. Otherwise, it looks good. We'll test it. Reviewing the algorithm will need some more time.